### PR TITLE
r-rpostgresql: There is a cpu that fails to install because config.gu…

### PR DESCRIPTION
**Pull request by h-denpo**
There is a cpu that fails to install because config.guess is old.
Update config.guess to the latest version.

### CODE REVIEW

**Comment by adamjstewart** [comment]
@alalazo we do this by default for AutotoolsPackage, is there any way to reuse the logic in an RPackage?

**Comment by alalazo** [comment]
I'll have a look at this in the next few days.

**Comment by alalazo** [comment]
Just a heads-up that I didn't forget this, only other things are taking more time than originally planned :sweat_smile:

File: var/spack/repos/builtin/packages/r-rpostgresql/package.py [hunk 1]
```diff
@@ -27,3 +27,4 @@ class RRpostgresql(RPackage):
     depends_on('r@2.9.0:', type=('build', 'run'))
     depends_on('r-dbi@0.3:', type=('build', 'run'))
     depends_on('postgresql')
+    patch('config.guess.patch')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/r-rpostgresql/package.py]: I think we may be able to use the following and remove the patch. Can you try if this:
```suggestion
    
    depends_on('automake', type='build')
    
    patch_config_files = True

    @run_before('install')
    def patch_config_guess(self):
        AutotoolsPackage._do_patch_config_files(self)

```
works for you? I currently don't have access to any `aarch64` machine to verify it.

**Review by alalazo — CHANGES_REQUESTED** [review]
I think we can try to reuse the code in `AutotoolsPackage` instead of committing a 4k lines patch. I don't have currently an `aarch64` machine to try what I suggest on, so beware that modifications may be needed to have it run correctly.

Ping me back if you encounter any trouble with this approach

File: var/spack/repos/builtin/packages/r-rpostgresql/package.py (hunk 1 above)
Comment (h-denpo) [reply var/spack/repos/builtin/packages/r-rpostgresql/package.py]: I tried your fix on aarch64 and it completed the installation without errors.

**Review by adamjstewart — APPROVED** [review]

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
Thanks @h-denpo !

**Pull request merged**